hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/test/java/org/apache/hadoop/mapreduce/TestJobSubmissionFiles.java

```diff
@@ -139,4 +141,19 @@ public void testGetStagingDirWhenShortFileOwnerNameAndShortUserName()
     assertEquals(stagingPath,
         JobSubmissionFiles.getStagingDir(cluster, conf, user));
   }
+
+  @Test
+  public void testDirPermission() throws Exception {
+      Cluster cluster = mock(Cluster.class);
+      HdfsConfiguration CONF = new HdfsConfiguration();
+      MiniDFSCluster cluster1 = new MiniDFSCluster.Builder(CONF).numDataNodes(2).build();
```

Comment:
Suggestion: different variable name for greater clarity, e.g. `dfsCluster`.

Also, please use a `finally` block to guarantee calling `MiniDFSCluster#shutdown()` at the end of the test.